auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthContext.java

```diff
@@ -13,6 +15,19 @@
  */
 public interface VaadinAuthContext {
 
+    /**
+     * Gets an {@link Optional} with an instance of the current user if it has
+     * been authenticated by a OpenID Connect provider, or empty if the user is
+     * not authenticated or has been authenticated by other means.
+     *
+     * @param userType
+     *            the type of the expected user instance, for example
+     *            {@link OidcUser} or a custom subclass
+     * @return an {@link Optional} with the current OIDC authenticated user, or
+     *         empty if none available
+     */
+    <U> Optional<U> getAuthenticatedUser(Class<? extends U> userType);
```

Comment:
Was it intended that the U has no bound declared as OidcUser allowing people to supply anything with the consequences that things might just fail later on? Wouldn't a compile error be better suited?

Reply:
I'm not sure what use cases are planned for the auth package - if it is planned to be used without Oauth2/OIDC in the future you are absolutely right! Otherwise the bound looks more safe :)